src/main/java/org/openpnp/machine/reference/vision/ReferenceFiducialLocator.java

```diff
@@ -177,40 +183,49 @@ public Location getHomeFiducialLocation(Location location, Part part) throws Exc
      * @return
      * @throws Exception
      */
-    private static Location getFiducialLocation(BoardLocation boardLocation, Placement fid)
-            throws Exception {
+	private static Location getFiducialLocation(BoardLocation boardLocation, Placement fid, Length fidDia) throws Exception {
         Camera camera = Configuration.get().getMachine().getDefaultHead().getDefaultCamera();
 
         Logger.debug("Locating {}", fid.getId());
 
+		Footprint footprint;
+
+		// Careful...magic number below. Also used in JobPanel. This are for
+		// global panel fiducials, which do not have an entry in the PCB BOM.
+		if ((fid.getId().equals("PanelFid1_a322") || fid.getId().equals("PanelFid2_g301"))) {
```

Comment:
I'm stuck on this part that you wrote: "BoardLocation gets panelId which refers to a Panel in Job.panels. We'll also materialize a BoardLocation.panel property during job load so that BoardLocation references can get you back to the Panel."

This suggests to me that BoardLocation will have both a panelId string and a panel property and that at load time the panelId will be turned into a panel instance, is that correct? If so, where precisely does that take place during job load? 

What I'm a bit confused on is if the BoardLocation knows the panel ID, it could just find the panel instance at when asked to locateboard(), yes? But how precisely?

Thanks